python/cugraph/cugraph/dask/structure/mg_property_graph.py

```diff
@@ -578,10 +577,6 @@ def get_vertex_data(self, vertex_ids=None, types=None, columns=None):
             if vertex_ids is not None:
                 if isinstance(vertex_ids, int):
                     vertex_ids = [vertex_ids]
-                elif not isinstance(
-                    vertex_ids, (list, slice, np.ndarray, self.__series_type)
-                ):
-                    vertex_ids = list(vertex_ids)
                 df = df.loc[vertex_ids]
```

Comment:
Removing the conversion to a list seems reasonable, esp. since it's slow and doing it completely means adding so many other types that would need to be converted.

The tradeoff is that we're allowing potentially bad types to be passed through to `.loc`.  Rather than converting things to a list, could we just catch `TypeError` and raise something a little more meaningful so it doesn't look like we have a bug if they pass in an incompatible type?

There is the risk that catching this could mask a cuDF bug (eg. a bug which results in a TypeError they didn't intentionally raise), but that seems less likely, and from the end-user's perspective is no different (ie. they'd file a bug report either way).
```suggestion
                try:
                    df = df.loc[vertex_ids]
                except TypeError:
                    raise TypeError(f"vertex_ids needs to be a list-like type compatible with DataFrame.loc[], got {type(vertex_ids)}")
```